test/contracts/core/modules/erc20RebalancingSetExchangeIssuanceModule.spec.ts

```diff
@@ -0,0 +1,912 @@
+    let exchangeIssueSendTokens: Address[];
+    let exchangeIssueSendTokenAmounts: BigNumber[];
+    let exchangeIssueReceiveTokens: Address[];
+    let exchangeIssueReceiveTokenAmounts: BigNumber[];
+
+    let zeroExOrder: ZeroExSignedFillOrder;
+
+    beforeEach(async () => {
+      // Create component token (owned by 0x order maker)
+      baseSetComponent = await erc20Wrapper.deployTokenAsync(zeroExOrderMaker);
+
+      // Create the Set (1 component)
+      const componentAddresses = [baseSetComponent.address];
+      const componentUnits = [new BigNumber(10 ** 10)];
+      baseSetNaturalUnit = new BigNumber(10 ** 9);
+      baseSetToken = await coreWrapper.createSetTokenAsync(
+        core,
+        setTokenFactory.address,
+        componentAddresses,
+        componentUnits,
+        baseSetNaturalUnit,
+      );
+
+      // Create the Rebalancing Set
+      rebalancingUnitShares = new BigNumber(10 ** 10);
+      rebalancingSetToken = await rebalancingWrapper.createDefaultRebalancingSetTokenAsync(
+        core,
+        rebalancingSetTokenFactory.address,
+        ownerAccount,
+        baseSetToken.address,
+        ONE_DAY_IN_SECONDS,
+        rebalancingUnitShares,
+      );
+
+      paymentToken = await erc20Wrapper.deployTokenAsync(tokenPurchaser);
+      subjectPaymentToken = paymentToken.address;
+
+      await erc20Wrapper.approveTransfersAsync(
+        [paymentToken],
+        transferProxy.address,
+        tokenPurchaser
+      );
+
+      subjectPaymentTokenAmount = customRequiredPaymentToken || new BigNumber(10 ** 10);
+
+      // Generate exchange issue data
+      exchangeIssueSetAddress = baseSetToken.address;
+      exchangeIssueQuantity = customExchangeIssueQuantity || new BigNumber(10 ** 10);
+      exchangeIssueSendTokenExchangeIds = [SetUtils.EXCHANGES.ZERO_EX];
+      exchangeIssueSendTokens = [subjectPaymentToken];
+      exchangeIssueSendTokenAmounts = [customIssuePaymentTokenAmount || subjectPaymentTokenAmount];
+      exchangeIssueReceiveTokens = componentAddresses;
+      exchangeIssueReceiveTokenAmounts = componentUnits.map(
+        unit => unit.mul(exchangeIssueQuantity).div(baseSetNaturalUnit)
+      );
+
+      const exchangeIssuanceParams = {
+        setAddress: 			      exchangeIssueSetAddress,
+        sendTokenExchangeIds:   exchangeIssueSendTokenExchangeIds,
+        sendTokens:             exchangeIssueSendTokens,
+        sendTokenAmounts:       exchangeIssueSendTokenAmounts,
+        quantity:               exchangeIssueQuantity,
+        receiveTokens:       	  exchangeIssueReceiveTokens,
+        receiveTokenAmounts: 	  exchangeIssueReceiveTokenAmounts,
+      };
+
+      await erc20Wrapper.approveTransfersAsync(
+        [baseSetComponent],
+        SetTestUtils.ZERO_EX_ERC20_PROXY_ADDRESS,
+        zeroExOrderMaker
+      );
+
+      // Create 0x order for the component, using weth(4) paymentToken as default
+      zeroExOrder = await setUtils.generateZeroExSignedFillOrder(
```

Comment:
same as before, this test is not very indicative of what we're going to launch with.
1. the set has a single component
2. it's a single 0x order
3. it's 0x
at the very minimum lets get these switched over to using kyber for the one component

Reply:
we have a second to breath now, we should circle back and address this the right way this time.